x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -471,24 +473,36 @@ export class SessionIndex {
               0
             );
             if (errorCount < bulkResponse.items.length) {
-              this.options.logger.warn(
+              logger.warn(
                 `Failed to clean up ${errorCount} of ${bulkResponse.items.length} invalid or expired sessions. The remaining sessions were cleaned up successfully.`
               );
             } else {
-              this.options.logger.error(
+              logger.error(
                 `Failed to clean up ${bulkResponse.items.length} invalid or expired sessions.`
               );
             }
           } else {
-            this.options.logger.debug(
-              `Cleaned up ${bulkResponse.items.length} invalid or expired sessions.`
-            );
+            logger.debug(`Cleaned up ${bulkResponse.items.length} invalid or expired sessions.`);
           }
         }
       }
     } catch (err) {
-      this.options.logger.error(`Failed to clean up sessions: ${err.message}`);
-      throw err;
+      logger.error(`Failed to clean up sessions: ${err.message}`);
+      error = err;
+    }
+
+    if (haveDeletedSessions) {
+      try {
+        await elasticsearchClient.indices.refresh({ index: this.indexName });
+        logger.debug(`Refreshed session index.`);
+      } catch (err) {
+        logger.error(`Failed to refresh session index: ${err.message}`);
+      }
+    }
```

Comment:
**nit:** Maybe put this block in a finally statement if you want it executed irregardless of  outcome? Thinks it's cleaner and you don't need to assign the thrown error.

Reply:
The old code threw the error in the `catch`. If we just tacked on a `finally`, it wouldn't get executed in that case, would it?